File: data/data_transpiled/at_coder/abc017/C/3874444.py
def import import sys , math , out
from os import urandom
import random
class Main ( object ) :
    def __init__ ( self ) :
        self.n = random.randint ( 1 , 3 )
        self.m = random.randint ( 1 , 3 )
        self.score = [ 0 ] * ( m + 2 )
        sum = 0
        for i in range ( n ) :
            l , r , s = random.randint ( 1 , 3 )
            score [ l ] += s
            score [ r + 1 ] -= s
            sum += s
        for i in range ( 1 , m + 1 ) :
            score [ i ] += score [ i - 1 ]
        ans = 0
        for i in range ( 1 , m + 1 ) :
            ans = max ( ans , sum - score [ i ] )
        out.write ( ans )

File: storage/data_original/at_coder/abc017/C/3874444.java
import java.util.*;
import static java.lang.System.*;
import static java.lang.Math.*;
public class Main {
    public static void main(String[]$) {
        Scanner sc = new Scanner(in);
        //??:???????i?????????????????score[]???????????????????????
        //??i????????????i???????????????0????????????????????????????
        int n=sc.nextInt();
        int m=sc.nextInt();
        int[] score=new int[m+2];
        int sum=0;
        for (int i = 0; i <n; i++) {
            int l=sc.nextInt(),r=sc.nextInt(),s=sc.nextInt();
            score[l]+=s;
            score[r+1]-=s;
            sum+=s;
        }
        for (int i = 1; i <=m; i++) {
            score[i]+=score[i-1];
        }
        int ans=0;
        for (int i = 1; i <=m; i++) {
            ans=max(ans,sum-score[i]);
        }
        out.println(ans);
    }
}